yt_dlp/extractor/nytimes.py

```diff
@@ -124,139 +82,408 @@ class NYTimesIE(NYTimesBaseIE):
         'only_matching': True,
     }]
 
-    def _extract_podcast_from_json(self, json, page_id, webpage):
-        podcast_audio = self._parse_json(
-            json, page_id, transform_source=js_to_json)
+    def _extract_content_from_block(self, block):
+        details = traverse_obj(block, {
+            'id': ('sourceId', {str}),
+            'uploader': ('bylines', ..., 'renderedRepresentation', {str}),
+            'duration': (('duration', 'length', ...), {int_or_none}),
+            'timestamp': ('firstPublished', {parse_iso8601}),
+            'series': ('podcastSeries', {str}),
+        }, get_all=False)
 
-        audio_data = podcast_audio['data']
-        track = audio_data['track']
+        formats, subtitles = self._extract_media_from_json(details.get('id'), block)
 
-        episode_title = track['title']
-        video_url = track['source']
+        # audio articles will have an url and no formats
+        url = traverse_obj(block, ('fileUrl', {url_or_none}))
+        if not formats and url:
+            formats.append({'url': url, 'ext': determine_ext(url)})
 
-        description = track.get('description') or self._html_search_meta(
-            ['og:description', 'twitter:description'], webpage)
+        return {
+            **details,
+            'formats': formats,
+            'subtitles': subtitles
+        }
 
-        podcast_title = audio_data.get('podcast', {}).get('title')
-        title = ('%s: %s' % (podcast_title, episode_title)
-                 if podcast_title else episode_title)
+    def _extract_thumbnails(self, thumbs):
+        return traverse_obj(thumbs, (lambda _, v: url_or_none(v['url']), {
+            'url': 'url',
+            'width': ('width', {int_or_none}),
+            'height': ('height', {int_or_none}),
+        }))
 
-        episode = audio_data.get('podcast', {}).get('episode') or ''
-        episode_number = int_or_none(self._search_regex(
-            r'[Ee]pisode\s+(\d+)', episode, 'episode number', default=None))
+    def _real_extract(self, url):
+        page_id = self._match_id(url)
+        webpage = self._download_webpage(url, page_id)
+        article_json = self._search_json(
+            r'window\.__preloadedData\s*=', webpage, 'media details', page_id,
+            transform_source=lambda x: x.replace('undefined', 'null'))['initialData']['data']['article']
+
+        blocks = traverse_obj(article_json, (
+            'sprinkledBody', 'content', ..., ('ledeMedia', None),
+            lambda _, v: v['__typename'] in ('Video', 'Audio')))
+
+        art_title = remove_end(self._html_extract_title(webpage), ' - The New York Times')
+        art_description = traverse_obj(
+            article_json, ('sprinkledBody', 'content', ..., 'summary', 'content', ..., 'text'),
+            get_all=False) or self._html_search_meta(['og:description', 'twitter:description'], webpage)
+        art_upload_date = traverse_obj(article_json, ('firstPublished'))
```

Comment:
inline below -- as I said in another comment, manually calculating `upload_date` from the same value that you are using for `timestamp` is just unnecessary complexity.

Also, `timestamp` is not being added to the entries. Is this intentional? If it was an oversight, then it might be better to construct a `common_info` dict here and unpack it in the entries and single video result below